### PR TITLE
Fix field defs for Document of  forge-viewer

### PR DESCRIPTION
**Pull request by yiskang**
Fix incorrect definitions of Document

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://forge.autodesk.com/en/docs/viewer/v7/change_history/changelog_v7/migration_guide_v6_to_v7/
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

**Comment by yiskang** [comment]
@alansmithnbs @liskaj Could you allocate time to review the PR? This is a request from Forge Help case no. 6784. Thank you very much!

File: types/forge-viewer/index.d.ts [hunk 1]
```diff
@@ -698,7 +700,7 @@ declare namespace Autodesk {
         }
 
         class Viewer3D {
-            constructor(container: HTMLElement, config?: Viewer3DConfig);
+            constructor(container: HTMLElement | null, config?: Viewer3DConfig);
```
[the remark this answers is not in the document]
Comment (yiskang) [reply types/forge-viewer/index.d.ts:703]: The container cannot be empty, but if I didn’t do this, the tslint will keep stopping at that line and warning the argument should allow null...

File: types/forge-viewer/index.d.ts [hunk 2]
```diff
@@ -21,7 +21,9 @@
 //
 /// <reference types="THREE" />
 
-declare namespace Autodesk {
+import * as THREE from 'three';
+
+export namespace Autodesk {
```
Comment (liskaj) [inline types/forge-viewer/index.d.ts:26]: I'm not sure about this. It's possible but if I try it in existing project I'm actually getting issues later when building the application (i.e. webpack is looking for forge-viewer npm module which obviously doesn't exist).

File: types/forge-viewer/index.d.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (liskaj) [reply types/forge-viewer/index.d.ts:703]: OK, can't we then change rules for tslint?

[the remark this answers is not in the document]
Comment (yiskang) [reply types/forge-viewer/index.d.ts:703]: Ok, I found a way to revert this change.

File: types/forge-viewer/index.d.ts (hunk 2 above)
Comment (yiskang) [reply types/forge-viewer/index.d.ts]: Found a way to revert the `declare namespace Autodesk`, please have a try!

**Comment by liskaj** [comment]
Looks fine to me.

**Review by liskaj — APPROVED** [review]

File: types/forge-viewer/tslint.json [hunk 3]
```diff
@@ -1,3 +1,6 @@
 {
-    "extends": "dtslint/dt.json"
+    "extends": "dtslint/dt.json",
+    "rules": {
+        "strict-export-declare-modifiers": false
```
Comment (weswigham) [inline types/forge-viewer/tslint.json:4]: Considering you don't change any export modifiers (anymore?) you should be able to revert this change to the `tslint.json` file.

**Review by weswigham — CHANGES_REQUESTED** [review]

File: types/forge-viewer/tslint.json (hunk 3 above)
Comment (yiskang) [reply types/forge-viewer/tslint.json:4]: I keep the original format for our users, but to keep the module declaration, I have turn off this tslint rule.

Comment (yiskang) [reply types/forge-viewer/tslint.json:4]: @weswigham If I use `export namespace` syntax instead, that will break lots of existing apps of our Forge viewer users.  Therefore, I have to omit this rule `strict-export-declare-modifiers` in the new commit. Should I revert the change?

Comment (weswigham) [reply types/forge-viewer/tslint.json:4]: ? What's this about `export namespace`? This rule is just about `declare` modifiers...? From what I can see, this PR is a one-line change to add a `static` modifier + a new test...?

Comment (yiskang) [reply types/forge-viewer/tslint.json:4]: Yes, according to the contribution guide, I have to write a test file for any changes, but the tslint will throw an error if I use the module to declare the typing. 

The ’static’ one is the request form our paid customer of the Autodesk Forge platform. Without this, it breaks our customers’ app while compiling the TypeScript to Javascript. Please don't block it to make Autodesk lose customers. We cannot take any risk to lose that one. Thanks!

Comment (yiskang) [reply types/forge-viewer/tslint.json]: A few days ago, while running the tslint, it asked to change the declaration of the forge-viewer module as below, I didn't know why.

```
// From
declare namespace Autodesk {
    namespace Viewing {
    }
}

// To
import * as THREE from 'three';

export namespace Autodesk {
    namespace Viewing {
    }
}
```

After @liskaj reviewed the PR, I found a way to revert back the `declare namespace Autodesk`, but it needed to omit the rule `strict-export-declare-modifiers` in the tslint. That's why I added the second commit. Now I amended the first commit with the changes I made in the second one. I didn't know why it passed now with omitting the rule.

Please don't block this PR, this request is from our paid Forge platform users. Autodesk cannot afford any risk to lose the customer. Thanks!

Comment (weswigham) [reply types/forge-viewer/tslint.json]: Right, there's no issues with the changes to the types or the tests as they currently stand, I'm just asserting that I thought you've reverted the changes that required you to include this lint suppression over the course of this PR (which is borderline trivial at this point), and it does not need to be here anymore, from what I can tell, as it's not suppressing any failures (though if it did I'd have suggested a different way of stating the types to avoid the suppression). If I'm mistaken and the rule's a new addition that causes failures in the current forge types (and simply hadn't been run before), I'd prefer an inline suppression on the failing line to a global one. (Given that this is a declaration file, it's probably just requesting you drop a redundant `declare` modifier if that is the case - you needn't replace it with `export` - everything in a declaration file is implicitly `declare`'d)

**Review by weswigham — APPROVED** [review]

**Comment by yiskang** [comment]
Ready to merge

File: types/forge-viewer/tslint.json (hunk 3 above)
Comment (yiskang) [reply types/forge-viewer/tslint.json]: Thank you very much for sharing your opinion. It seems to me that there are lots of things to learn. I will try inline suppression next time. Cheers!
